mslice/app/__init__.py

```diff
@@ -20,6 +23,8 @@ def show_gui():
     If this is the first call then an instance of the Windows is cached to ensure it
     survives for the duration of the application
     """
+    AlgorithmFactory.subscribe(MakeProjection)
+    _translate()
```

Comment:
@martyngigg I had to add the `_translate()` call here for `simpleapi` to pick up the algorithm... is there a better way of doing this?

Reply:
@MatthewBowles You're right - this is a little messy. The first thing I would do is have a separate module and function to handle all of the registration. Maybe a `mslice.util.mantid` with an `initialize_mantid` or something? We could also then move the `run_alg` function into there.

For the `_translate` step, how about calling `_create_algorithm_function` directly? That way we avoid having to run through all of the algorithms again?

Reply:
This is better. Apologies for the cluttered diff, I had to split `workspace_provider` to avoid dependency problems.